src/trace-plugin-loader.js

```diff
@@ -159,6 +163,11 @@ function activate(agent) {
         if (patchedRoot !== null) {
           return patchedRoot;
         }
+      } else {
+        var modulePath = Module._resolveFilename(request, parent).replace('/', path.sep);
```

Comment:
The issue is with the persistence of the patch/intercept. Patches persist because the object that's being mutated is being cached, so the original module loader should be able to retrieve it. On the other hand, the only indication that a module was intercepted is within the scope of this file.